cordapp/src/main/kotlin/net/corda/plugins/StrippedMavenPom.kt

```diff
@@ -0,0 +1,152 @@
+package net.corda.plugins;
+
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.XmlProvider
+import org.gradle.api.provider.Property
+import org.gradle.api.publish.maven.*
+import org.gradle.api.publish.maven.internal.dependencies.MavenDependencyInternal
```

Comment:
Internal Gradle interfaces and APIs? Does this impose any version compatibility constraints on our uses of Gradle between Open Source, Enterprise and our Gradle Plugins projects?

Reply:
In Maven's infinite wisdom - you *have* to use the Internal version. Because if you do not use the internal version of things, you get a nasty class cast error. 

I mean, look at this:
https://github.com/gradle/gradle/blob/42a750cf566da31cda1bf18437abaf3c40fcc659/subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/GenerateMavenPom.java#L119